fsspec/implementations/sftp.py

```diff
@@ -64,9 +64,20 @@ def _get_kwargs_from_urls(urlpath):
         out.pop("protocol", None)
         return out
 
-    def mkdir(self, path, mode=511):
-        logger.debug("Creating folder %s" % path)
-        self.ftp.mkdir(path, mode)
+    def mkdir(self, path, create_parents=False, mode=511):
+        logger.info("Creating folder %s" % path)
+        if self.exists(path):
+            raise FileExistsError("File exists: {}".format(path))
```

Comment:
It seems unnecessary to make this extra check, we would expect the remote to give a reasonable error when you try to create the dir.

Reply:
I removed this extra check, then test are failed due to `FileExistsError`. That's why I reverted it, now all tests are successful on my end.

Reply:
Can we catch the paramiko exception and turn it into FileExists without first checking?
```
try:
...
    self.ftp.mkdir(path, mode)
except paramiko-exists-error as e:
    raise FileExistsError from e
except paramiko-path-doesn't-exist-error as e:
    raise FileNotFound from e
```

Reply:
In [paramiko>sftp_client.py:891](https://github.com/paramiko/paramiko/blob/29c4d3210b745e7d56c328ffebc2513ebfb482fb/paramiko/sftp_client.py#L891), it already raises `FileNotFoundError` and `Failure` for `FileExistsError` (other exceptions included too) . If we create this type of control we double the `FileNotFoundError` and miss the exact `FileExistsError` since no identifier presented for it in the `paramiko`.
I think it's better to check FileExists in there and let `pyarrow` control the exception. Check [pyarrow>fs.py:339](https://github.com/apache/arrow/blob/306d0b4144527fb988da3b64ec74b53536d974f0/python/pyarrow/fs.py#L339)

Reply:
`pyarrow` ? You have confused me :|

Reply:
Sorry:) I'm considering this scenario, that's why I guess:

> An example issue appears when these implementation classes are used together with pyarrow's PyFileSystem. FSSpecHandler translates its create_dir(self, path, recursive) calls into mkdir(path, create_parents=recursive) following AbstractFileSystem standards and it fails due to missing parameter in implementation.